src/raft/recv_install_snapshot.c

```diff
@@ -8,6 +8,348 @@
+	int result;
+};
+
+/**
+ * =Operation
+ *
+ * 0. Leader creates one state machine per Follower to track of their
+ * states and moves it to FOLLOWER_ONLINE state. Follower creates a
+ * state machine to track of its states and moves it to NORMAL state.
+ *
+ * 1. The Leader learns the Follower’s follower.lastLogIndex during
+ * receiving replies on AppendEntries() RPC, fails to find
+ * follower.lastLogIndex in its RAFT log or tries and fails to
+ * construct an AppendEntries() message because of the WALs that
+ * contained some necessary frames has been rotated out, and
+ * understands that the snapshot installation procedure is
+ * required.
+ *
+ * Leader calls leader_tick() putting struct raft_message as a
+ * parameter which logic moves it from FOLLOWER_ONLINE to
+ * FOLLOWER_NEEDS_SNAPSHOT state.
+ *
+ * 2. The Leader initiates snapshot installation by sending
+ * InstallSnapshot() message.
+ *
+ * Upon receiving this message on the Follower's side, Follower calls
+ * follower_tick() putting struct raft_message as a parameter which
+ * logic moves it from NORMAL to SIGNATURES_CALC_STARTED state.
+ *
+ * 3. The Follower calculates [page_checksum_t] for its local
+ * persistent state. When this process ends it puts itself into
+ * SIGNATURES_CALC_DONE state by calling follower_tick(..., NULL).
+ *
+ * 4. The Follower sends Signature() messages to the Leader, transfers
+ * the whole [page_checksum_t] and puts itself into
+ * SIGNATURES_PART_SENT state. When all signatures transferred the
+ * Leader, it sends Signature(done=true) message.
+ *
+ * 5. When the Leader received the first Signature() message it moves
+ * corresponding state machine into SIGNATURES_CALC_STARTED state and
+ * creates HT. The Leader puts incomming payloads of Signature()
+ * message into the HT. When the Leader received Signature(done=true)
+ * message it moves into SNAPSHOT_INSTALLATION_STARTED state.
+ *
+ * 6. In SNAPSHOT_INSTALLATION_STARTED state, the Leader starts
+ * iterating over the local persistent state, and calculates the
+ * checksum for each page the state has. Then, it tries to find the
+ * checksum it calculated in HT. Based on the result of this
+ * calculation, the Leader sends InstallShapshot(CP..) or
+ * InstallShapshot(MV..) to the Follower.
+ *
+ * Upon receving these messages the Follower moves into
+ * SNAPSHOT_CHUNCK_RECEIVED state. The Leader moves into
+ * SNAPSHOT_CHUNCK_SENT state after receiving first reply from the
+ * Follower.
+ *
+ * 7. When the iteration has finished the Leader sends
+ * InstallShapshot(..., done=true) message to the Follower. It moves
+ * the Follower back to NORMAL state and the state machine
+ * corresponding to the Follower on the Leader is moved to
+ * SNAPSHOT_DONE_SENT state.
+ *
+ * 8. The Leader sends AppendEntries() RPC to the Follower and
+ * restarts the algorithm from (1). The Leader's state machine is
+ * being moved to FOLLOWER_ONLINE state.
```

Comment:
Would it be more accurate to say that the Leader's state machine is moved to FOLLOWER_ONLINE once the response `InstallSnapshotResult` is received. From the text written in 8. it looks like it waits in FOLLOWER_ONLINE and sends AppendEntries() there.